wsagent/che-core-api-git-shared/src/main/java/org/eclipse/che/api/git/shared/Revision.java

```diff
@@ -20,34 +21,148 @@
 @DTO
 public interface Revision {
 
+    /**
+     * Parameter which shows that this revision is a fake revision (i.e. TO for Exception)
+     *
+     * @return boolean
+     */
+    boolean isFake();
+
+    /** @set boolean value which represents whether this revision is a fake revision*/
+    void setFake(boolean fake);
+
     /** @return branch name */
     String getBranch();
-    
+
+    /** @set branch name*/
     void setBranch(String branch);
-    
+
+    /**
+     * Create a Revision object based on a given branch name.
+     *
+     * @param branch
+     *         rbranch name
+     * @return a Revision object
+     */
     Revision withBranch(String branch);
 
     /** @return commit id */
     String getId();
-    
+
+    /** @set commit id*/
     void setId(String id);
-    
+
+    /**
+     * Create a Revision object based on a given commit id.
+     *
+     * @param id
+     *         commit id
+     * @return a Revision object
+     */
     Revision withId(String id);
 
     /** @return commit message */
     String getMessage();
-    
+
+    /** @set commit message*/
     void setMessage(String message);
-    
+
+    /**
+     * Create a Revision object based on a given commit message.
+     *
+     * @param message
+     *         commit message
+     * @return a Revision object
+     */
     Revision withMessage(String message);
 
     /** @return time of commit */
     long getCommitTime();
-    
+
+    /** @set commit time*/
+    void setCommitTime(long time);
+
+    /**
+     * Create a Revision object based on a given commit time.
+     *
+     * @param time
+     *         commit time
+     * @return a Revision object
+     */
     Revision withCommitTime(long time);
 
-    /** @return committer */
+    /** @return GitUser object which represents the commit committer */
     GitUser getCommitter();
-    
-    Revision withCommitter(GitUser user);
-}
\ No newline at end of file
+
+    /** @set GitUser object which represents the commit committer*/
+    void setCommitter(GitUser committer);
+
+    /**
+     * Create a Revision object based on a given GitUser object which represents the commit committer
```

Comment:
Create a Revision object based on a given GitUser object which represents the committer